ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx

```diff
@@ -38,31 +51,275 @@ export function showErrorMessages(errors: any) {
+      }
+      if (!bufferWidth || bufferWidth == 0) {
+        errors.add('Line buffer width must be greater than 0.000001')
+      }
+      break
+    case 'Point':
+      if (!bufferWidth || bufferWidth < 0.000001) {
+        errors.add('Radius must be greater than 0.000001')
+      }
+      if (
+        geometry.coordinates.some(
+          (coord: any) => !coord || coord.toString().length == 0
+        )
+      ) {
+        errors.add('Coordinates must not be empty')
+      }
+      break
+    case 'BoundingBox':
+      const box = filter.geojson.properties
+      if (!box.east || !box.west || !box.north || !box.south) {
+        errors.add('Bounding Box must have valid values')
+      }
+      break
   }
   return errors
 }
+
+function validateDDLatLon(label: string, defaultCoord: number, value: string) {
+  let message = ''
+  let defaultValue
+  if (value !== undefined && value.length === 0) {
+    message = `${label.replace(/^\w/, c => c.toUpperCase())} cannot be empty`
+    return { error: true, message, defaultValue }
+  }
+  if (Number(value) > defaultCoord || Number(value) < -1 * defaultCoord) {
+    defaultValue = Number(value) > 0 ? defaultCoord : -1 * defaultCoord
+    message = getDefaultingErrorMessage(value, label, defaultValue)
+    return { error: true, message, defaultValue }
+  }
+  return { error: false, message, defaultValue }
+}
+
+function validateDmsLatLon(label: string, value: string) {
+  let message = ''
+  let defaultValue
+  const validator = label === 'latitude' ? 'dd°mm\'ss.s"' : 'ddd°mm\'ss.s"'
+  if (value !== undefined && value.length === 0) {
+    message = `${label.replace(/^\w/, c => c.toUpperCase())} cannot be empty`
+    return { error: true, message, defaultValue }
+  }
+  if (validateInput(value, validator) !== value) {
+    defaultValue = validateInput(value, validator)
+    message = getDefaultingErrorMessage(value, label, defaultValue)
+    return { error: true, message, defaultValue }
+  }
+  return { error: false, message, defaultValue }
+}
+
+function validateUsng(value: string) {
+  if (value === '') {
+    return { error: true, message: 'USNG / MGRS coordinates cannot be empty' }
+  }
+  const result = converter.USNGtoLL(value, true)
+  const isInvalid = Number.isNaN(result.lat) || Number.isNaN(result.lon)
+  return {
+    error: isInvalid,
+    message: isInvalid ? 'Invalid USNG / MGRS coordinates' : '',
+  }
+}
+
+function upsValidDistance(distance: number) {
+  return distance >= 800000 && distance <= 3200000
+}
+
+function validateLatLon(lat: string, lon: string) {
+  const latitude = parseFloat(lat)
+  const longitude = parseFloat(lon)
+  return latitude > -90 && latitude < 90 && longitude > -180 && longitude < 180
+}
+
+function validateUtmUps(
+  key: string,
+  utmUpsEasting: string,
+  utmUpsNorthing: string,
+  zoneNumber: any,
+  hemisphere: any
+) {
+  let error = { error: false, message: '' }
+  zoneNumber = Number.parseInt(zoneNumber)
+  hemisphere = hemisphere.toUpperCase()
```

Comment:
could get rid of this line and put `hemisphere.toUpperCase()` in the same line as ` const northernHemisphere = hemisphere === 'NORTHERN'`. and since the converter doesn't even use the `hemisphere` variable it could be eliminated from `utmUpsParts` (I'm like 90% sure)